core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BoundedRangeFileInputStream.java

```diff
@@ -62,12 +62,15 @@ public BoundedRangeFileInputStream(FSDataInputStream in, long offset, long lengt
 
   @Override
   public int available() throws IOException {
-    int avail = in.available();
-    if (pos + avail > end) {
-      avail = (int) (end - pos);
-    }
+    final FSDataInputStream inLocal = in;
+    synchronized (inLocal) {
+      int avail = in.available();
```

Comment:
According to InputStream, this returns an estimate of the number of bytes that can be read (or skipped over) without blocking by the next invocation of a method for this input stream.  A single read or skip of this many bytes will not block, but may read or skip fewer bytes.
So we have to delegate this to the underlying stream.